docs/content/preview/reference/drivers/csharp/yb-npgsql-reference.md

```diff
@@ -0,0 +1,201 @@
+| TrustServerCertificate | For use with the Require SSL mode |
+
+Npgsql supports SSL modes in different ways depending on the driver version, as shown in the following table.
+
+| SSL mode | Versions before 6.0 | Version 6.0 or later |
+| :------- | :------------------------ | :----|
+| Disable  | Supported (default) | Supported |
+| Allow    | Not Supported | Supported |
+| Prefer   | Supported | Supported (default) |
+| Require  | Supported<br/>For self-signed certificates, set `TrustServerCertificate` to true | Supported<br/>Set `TrustServerCertificate` to true |
+| VerifyCA | Not Supported - use Require | Supported |
+| VerifyFull | Not Supported - use Require | Supported |
+
+The .NET Npgsql driver validates certificates differently from other PostgreSQL drivers as follows:
+
+- Prior to version 6.0, when you specify SSL mode `Require`, you also need to specify `RootCertificate`, and the driver verifies the certificate by default (like the verify CA or verify full modes on other drivers), and fails for self-signed certificates.
+
+  To use self-signed certificates, specify `TrustServerCertificate=true`, which bypasses walking the certificate chain to validate trust and hence works like other drivers' require mode. In this case, you don't need to specify the `RootCertificate`.
+
+- For version 6.0 and later, the `Require` SSL mode requires explicitly setting the `TrustServerCertificate` field to true.
+
+The following example shows how to build a connection string for connecting to a YugabyteDB cluster using the `Require` SSL mode.
+
+```csharp
+var connStringBuilder = new NpgsqlConnectionStringBuilder();
+    connStringBuilder.Host = "22420e3a-768b-43da-8dcb-xxxxxx.aws.ybdb.io";
+    connStringBuilder.Port = 5433;
+    connStringBuilder.SslMode = SslMode.Require;
+    connStringBuilder.Username = "admin";
+    connStringBuilder.Password = "xxxxxx";
+    connStringBuilder.Database = "yugabyte";
+    connStringBuilder.TrustServerCertificate = true;
+    CRUD(connStringBuilder.ConnectionString);
+```
+
+The following example shows how to build a connection string for connecting to a YugabyteDB cluster using the `VerifyCA` and `VerifyFull` SSL modes.
+
+```csharp
+var connStringBuilder = new NpgsqlConnectionStringBuilder();
+    connStringBuilder.Host = "22420e3a-768b-43da-8dcb-xxxxxx.aws.ybdb.io";
+    connStringBuilder.Port = 5433;
+    connStringBuilder.SslMode = SslMode.VerifyCA;
+    //or connStringBuilder.SslMode = SslMode.VerifyFull;
+    connStringBuilder.RootCertificate = "/root.crt"; //Provide full path to your root CA.
+    connStringBuilder.Username = "admin";
+    connStringBuilder.Password = "xxxxxx";
+    connStringBuilder.Database = "yugabyte";
+    CRUD(connStringBuilder.ConnectionString);
+```
+
+For more information on TLS/SSL support, see [Security and Encryption](https://www.npgsql.org/doc/security.html?tabs=tabid-1) in the Npgsql documentation. -->
+
+## Compatibility matrix
```

Comment:
@Sfurti-yb - Can you mention what versions we need to mention in this Compatibility Matrix.